### PR TITLE
proxy: use postgres_protocol scram/sasl code

### PR DESCRIPTION
**Pull request by conradludgate**
1) `scram::password` was used in tests only. can be replaced with `postgres_protocol::password`.
2) `postgres_protocol::authentication::sasl` provides a client impl of SASL which improves our ability to test

## Checklist before requesting a review

- [X] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

**Comment by arpad-m** [comment]
I'm not sure it is a good idea to replace a Rust dependency with a C one in a component as exposed as the proxy.

**Comment by conradludgate** [comment]
> I'm not sure it is a good idea to replace a Rust dependency with a C one in a component as exposed as the proxy.

What do you mean? postgres-protocol is a Rust fully library and we already indirectly depend on it through tokio-postgres.

> but to make sure one should make postgres-protocol a dev dependency.

I have direct plans to use postgres-protocol further soon (i plan on removing some uses of tokio-postgres), so I see no point to this

**Comment by arpad-m** [comment]
> postgres-protocol is a fully Rust library

Oh it is indeed, my concern is moot then. I'm still getting familiar to everything, I apologize.

> I have direct plans to use postgres-protocol further soon

fair

File: proxy/src/scram.rs [hunk 1]
```diff
@@ -121,4 +112,33 @@ mod tests {
             ]
         );
     }
+
+    fn run_round_trip_test(client_password: &str) {
+        let scram_secret = ServerSecret::build("pencil").unwrap();
```
Comment (khanova) [inline proxy/src/scram.rs:117]: Let's make it configurable in arguments.

**Review by khanova — APPROVED** [review]

**Pull request merged**